Update siglensent helm charts

venkateshamatam opened the pull request:
# Description 

Steps to deploy these Helm charts: 

1. Clone the repo and `cd charts`
2. Make sure to update `charts/siglensent/values.yaml` with the appropriate license key and host names. If you do not have a host name, you can use a self-signed certificate. 
3. Run `helm install siglensent ./siglensent -n siglensent -f ./siglensent/Values.yaml`
4. Verify that the services and pods are created:
  `kubectl get services -n siglensent` 
  `kubectl get pods -n siglensent`

AndrewHess commented on charts/siglensent/templates/raft-configmap.yaml:
```diff
@@ -0,0 +1,35 @@
+      dbname: db1
+    ingestNode: true
+    queryNode: true
+    etcd:
+      enabled: true
+      username: root
+      password: dontkeepthispassword
```
1. The username and password should be in values.yaml so it's easier for users to change (and because it's duplicated in the raft-configmap and the worker-configmap)
2. The `enabled` should also be in values.yaml, unless you have some other config in values.yaml that will make it so this raft-configmap.yaml doesn't do anything when the chart is used in a single-node configuration (since in that configuration, we don't need raft)

AndrewHess commented on charts/siglensent/templates/raft-configmap.yaml:
I'm not seeing `isRaftNode: true` anywhere. Am I just missing it, or do you not need it?

AndrewHess commented on charts/siglensent/templates/ingest-service.yaml:
```diff
@@ -0,0 +1,13 @@
+    app: {{ .Values.ingestService.selector.app }}
+  ports:
+    - protocol: TCP
+      port: 8081
+      targetPort: 8081
```
These should be configurable in values.yaml

Similar comment for query-service.yaml

AndrewHess commented on charts/siglensent/templates/ingress.yaml:
```diff
@@ -22,7 +22,7 @@ spec:
         backend:
           service:
-            name: {{ .Release.Name }}-query-service
+            name: query-service
```
Why don't you want this release name prefix? (It's been a while since I've used helm, so I might be forgetting something)

AndrewHess commented on charts/siglensent/templates/worker-configmap.yaml:
```diff
@@ -0,0 +1,30 @@
+      password: dontkeepthispassword
+      seedUrls:
+        - {{ printf "http://%s-etcd.%s.svc.cluster.local:2379" .Release.Name .namespace }}
+    raft:
+      enabled: false
```
This doesn't seem right. Does this work for a multinode cluster (e.g., sending a query gets all the data from the cluster, not just the data from one worker)? I expected something like:
```
isRaftNode: false
raft:
    enabled: true
    username: ...
    password: ...
    raftNodes: ...
```

venkateshamatam replied:
Yes it worked without `isRaftNode` set. I ingested different data into each worker node and running a query gets all data from the cluster.

AndrewHess commented on charts/siglensent/values.yaml:
```diff
@@ -8,55 +8,36 @@ ingress-nginx:
-image: 123456789012.dkr.ecr.us-east-1.amazonaws.com/siglens:latest
-imagePullPolicy: Always
-
-# If you're using an enterprise image, set "enterprise" to true. Enterprise
-# images require a license. To specify your license, set "licenseBase64" to the
-# output of `cat /path/to/licenseKey.txt | base64`.
-enterprise: false
-licenseBase64: ""
```
You replaced this with
```
secrets:
  licenseKey: ""
```
Should people using this chart set this to the raw text in the license file? Or to something else (e.g., the path to the file)? The license file has binary data, so there might be some issues with directly copy-pasting that here, which is why we were doing a base64 encoding previously. Did you not have issues with your new way?

venkateshamatam replied:
The new licenseKey should also be base64 encoded, I added the comment back. It's the same but just renamed from `licenseBase64` to "secrets.licenseKey" in `Values.yaml`

AndrewHess commented on charts/siglensent/values.yaml:
```diff
@@ -8,55 +8,36 @@ ingress-nginx:
-    rbac:
-      rootPassword: dontkeepthispassword
-
-# Used for pulling the SigLens image from a private registry. Not needed for
-# public registries. If you're using a private registry, set this to the output
-# of `cat ~/.docker/config.json | base64`.
-dockerConfigBase64: ""
```
You moved this to:
```
registry:
  dockerConfigBase64: ""
```
And that's fine, but why did you remove the comment? I think the comment will help clarify exactly what people using the chart should set the value to

AndrewHess reviewed (commented):
One note: there's quite a few values (ports, storage size, etc.) that you've hard coded inside a template file. These should all be in values.yaml and then those values should be read into the template. Basically, the templates files should give the basic structure of how we deploy to k8s, but if someone deploying the chart wants to make any change about specifics they should only need to modify values.yaml

venkateshamatam commented:
@AndrewHess  I addressed all comments, but with `isRaftNode: true` the worker-deployment pods fail to start with this error. There is no issue with raft-deployment pods.

```
time="2024-08-27 14:26:38" level=error msg="etcd.Init: Failed to get member list from etcd: context deadline exceeded"
time="2024-08-27 14:26:38" level=error msg="InitalizeDistributedStorage: Failed to initialize etcd storage: context deadline exceeded"
time="2024-08-27 14:26:38" level=error msg="Failed to init distributed storage: context deadline exceeded"
time="2024-08-27 14:26:38" level=warning msg="Flushing 0 segment files on server exit"
{"level":"warn","ts":"2024-08-27T14:26:38.124686Z","logger":"etcd-client","caller":"v3@v3.5.13/retry_interceptor.go:62","msg":"retrying of unary invoker failed","target":"etcd-endpoints://0xc0001b7680/raft-deployment-0.raft-service.siglensent.svc.cluster.local:2379","attempt":0,"error":"rpc error: code = DeadlineExceeded desc = latest balancer error: last connection error: connection error: desc = \"transport: Error while dialing: dial tcp 10.24.0.24:2379: connect: connection refused\""}
time="2024-08-27 14:26:38" level=warning msg="FlushAliasMapToFile: Flushing alias map to file on exit"
time="2024-08-27 14:26:38" level=info msg="ForcedFlushToScrollFile: Flushing scroll context to file"
time="2024-08-27 14:26:38" level=error msg="siglens main: Error in starting server:context deadline exceeded "
```

AndrewHess commented:
> @AndrewHess I addressed all comments, but with `isRaftNode: true` the worker-deployment pods fail to start with this error. There is no issue with raft-deployment pods. Could you confirm if the config is right or if something is missing?

The worker config looks correct now. Also I believe the reason you're only seeing this issue now is because now in the `raft` section you have `enabled: true` (which is correct); the issue isn't that you have `isRaftNode: false` (which is also correct). Before, when you had `enabled: false`, the workers weren't trying to connect to the raft cluster

AndrewHess commented on charts/siglensent/templates/raft-deployment.yaml:
```diff
@@ -0,0 +1,82 @@
+      - name: raft-container
+        image: {{ .Values.raft.deployment.image }}
+        imagePullPolicy: {{ .Values.raft.deployment.imagePullPolicy }}
+        command: ["/bin/sh"]
+        args:
+          - -c
+          - |
+            sleep 10
+            exec /siglens/sigscalr --config /config/server.yaml
```
Why is `sleep 10` necessary?

AndrewHess replied:
nit: in worker-deployment.yaml you start like this:
```
        command: ["/siglens/sigscalr"]
        args: ["--config", "/config/server.yaml"]
```
Why don't you start them both the same?

AndrewHess commented on charts/siglensent/Chart.yaml:
```diff
@@ -25,8 +25,4 @@ dependencies:
-    version: "10.0.3"
-    repository: https://charts.bitnami.com/bitnami
-    condition: multinode.enabled
+    repository: https://charts.jetstack.io
```
Removing the `etcd` chart is going to break multinode configured with S3. I'm pretty sure some other changes in this PR will also break S3 multinode.

@nkunal do we want to accept this PR with these breaking changes, or take a little more time to ensure we don't break the S3 configuration?

AndrewHess commented on charts/siglensent/values.yaml:
```diff
@@ -93,5 +88,33 @@ config:
+  worker:
+    ssInstanceName: "worker-node"
+
+raft:
+  deployment:
+    replicas: 3 # Number of replicas for the Raft server
+    image: public.ecr.aws/l3t9n5v5/siglens-enterprise:0.2.21-ent1
+    imagePullPolicy: Always
+    ports:
+      ingest: 8081
+      raft: 2379
+    storage:
+      size: "10Gi"
+
+worker:
+  deployment:
+    replicas: 3 # Number of replicas for the Worker server
+    image: public.ecr.aws/l3t9n5v5/siglens-enterprise:0.2.21-ent1
+    imagePullPolicy: Always
+    ports:
+      ingest: 8081
```
nit: I can't think of a reason someone would want a different image or pull policy for the Raft and Worker nodes. I think you should have those outside the `raft:` and `worker:` sections (maybe in the `registry:` section?), unless there's a specific reason you configure them twice.

AndrewHess replied:
nit: why does the `raft` section have `ingest: 8081`? Why does it need to know which port the workers use for ingest?

AndrewHess commented on charts/siglensent/values.yaml:
```diff
@@ -8,55 +8,50 @@ ingress-nginx:
-
-ebs:  # Used for sharing data between pods on the same node.
-  storageSize: 20Gi
+# images require a license. To specify your license, set "licenseBase64" to the
+# output of `cat /path/to/licenseKey.txt | base64`.
+secrets:
+  licenseKey: ""
```
nit: you should probably include `base64` in the `licenseKey` name, so it will be more clear. It used to be called `licenseBase64`

AndrewHess commented on charts/siglensent/templates/raft-configmap.yaml:
```diff
@@ -0,0 +1,30 @@
+    rbac:
+      provider: sqlite
+      dbname: db1
+    ingestNode: true
+    queryNode: true
+    isRaftNode: true
```
For raft nodes, you should have:
```
ingestNode: false
queryNode: false
```
(but keep `isRaftNode: true`)

AndrewHess commented on charts/siglensent/templates/raft-deployment.yaml:
```diff
@@ -0,0 +1,82 @@
+          - |
+            sleep 10
+            exec /siglens/sigscalr --config /config/server.yaml
+        ports:
+        - containerPort: {{ .Values.raft.deployment.ports.ingest }}
+        - containerPort: {{ .Values.raft.deployment.ports.raft }}  # Raft port
```
nits:
1. Why do you need the `ingest` port?
2. ` # Raft port` is already clear from the code, so you can delete this comment

AndrewHess commented on charts/siglensent/templates/raft-deployment.yaml:
```diff
@@ -0,0 +1,82 @@
+            path: license.txt
+      - name: cert-volume
+        secret:
+          secretName: ingress-tls
```
Somewhere else you looked up `ingress-tls` from values.yaml. Either look it up everywhere, or hardcode it everywhere.

I think a user of this chart would never have a reason to change `ingress-tls` to something else, is that right? If it is, I don't see a problem with hardcoding it. If it's not, you definitely want to have it configurable in values.yaml

AndrewHess commented on charts/siglensent/templates/raft-service.yaml:
```diff
@@ -0,0 +1,12 @@
+  clusterIP: None
+  selector:
+    app: raft-node
+  ports:
+    - port: 2379
+      targetPort: 2379
```
Don't you need to read these from values.yaml?

AndrewHess commented on charts/siglensent/templates/worker-deployment.yaml:
```diff
@@ -0,0 +1,58 @@
+            path: license.txt
+      - name: cert-volume
+        secret:
+          secretName: ingress-tls
```
Here you also hardcode `ingress-tls`

AndrewHess commented on charts/siglensent/values.yaml:
```diff
@@ -8,55 +8,50 @@ ingress-nginx:
 
-image: 123456789012.dkr.ecr.us-east-1.amazonaws.com/siglens:latest
-imagePullPolicy: Always
+etcd:
+  enabled: true
+  username: root
+  password: dontkeepthispassword
```
Why did you change this `etcd` config? It used to be:
```
# Overrides for the etcd subchart.
etcd:
  namespace: siglensent
  replicaCount: 3
  auth:
    rbac:
      rootPassword: dontkeepthispassword
```
And I think it's only used by the `etcd` chart which is used for multinode in S3 configuration. But you removed that chart in this PR

AndrewHess commented on charts/siglensent/values.yaml:
```diff
@@ -8,55 +8,50 @@ ingress-nginx:
+  port: 5122
+  targetPort: 5122
+
+ingress:
+  hosts:
+    - siglens.example.com
+    - ingest.siglens.example.com
+  tlsSecret: ingress-tls
```
You have some duplication. These URLs are what's specified a ways above:
```
queryHost: siglens.example.com
ingestHost: ingest.siglens.example.com
```
Can we remove this duplication? If not, it would be good to at least move these two sections close to each other so that if a user of this chart has to modify them, they'll be less likely to accidentally modify one section and forget to modify the other section

AndrewHess commented on charts/siglensent/values.yaml:
```diff
@@ -93,5 +88,30 @@ config:
+      ingest: 8081
+
+
+# Used for pulling the SigLens image from a private registry. Not needed for
+# public registries. If you're using a private registry, set this to the output
+# of `cat ~/.docker/config.json | base64`.
+registry:
+  dockerConfigBase64: ""
+  image: public.ecr.aws/l3t9n5v5/siglens-enterprise:0.2.21-ent1
+  imagePullPolicy: Always
```
nit: clarify that this comment is referring only to `dockerConfigBase64`. And maybe move the comment inside the `registry` block?
